### ISSUE TITLE
Consider baseColWidth when reading the XLSX file

### ISSUE THREAD

**Issue by Pankraty** [ClosedXML/ClosedXML#837]
## Read and complete the full issue template

**Do you want to request a *feature* or report a *bug*?**
- [x] Bug
- [ ] Feature

**Version of ClosedXML**

e.g. 0.93.0-beta2

**What is the current behavior?**

The issue #829 revealed another problem. Default column width may be specified in two different ways: using `defaultColWidth` and `baseColWidth` (see [documentation](https://msdn.microsoft.com/en-US/library/office/documentformat.openxml.spreadsheet.sheetformatproperties.aspx)). ClosedXML only considers `defaultColWidth` and if column width is specified using `baseColWidth` it is simply ignored. On saving or copying a worksheet this information is lost, and new columns are created with the default width (8.43).

**What is the expected behavior or new feature?**

In case `defaultColWidth` is not specified and `baseColWidth` is set we should set `XLWorksheet.ColumnWidth` accordingly (`baseColumnWidth + {margin padding (2 pixels on each side, totalling 4 pixels)} + {gridline (1pixel)}`)

**Did this work in previous versions of our tool?  Which versions?**

No

## Reproducibility
**This is an important section. Read it carefully. Failure to do so will cause a 'RTFM' comment.**

**Code to reproduce problem:**
```c#
public void Main()
{
            using (var wb = new XLWorkbook(@"829_sample.xlsx"))
            {
                var ws = wb.Worksheet(1);
                Console.WriteLine("Expected {0}, actual {1}", 11.17, ws.ColumnWidth); // actual 8.43
                Console.WriteLine("Expected {0}, actual {1}", 11.17, ws.Column(1).Width);  // actual 8.43
            }

            Console.WriteLine("Press any key to continue");
            Console.ReadKey();
}
```
- [x] I attached a sample spreadsheet.  (You can drag files on to this issue)

[829_sample.xlsx](https://github.com/ClosedXML/ClosedXML/files/1977595/829_sample.xlsx)

**Comment by igitur** [comment]
OK, are you going to fix it? Else I'll look at it.

**Comment by Pankraty** [comment]
The fix seems to be trivial, I almost finished. Although I am not sure about converting pixels to actual column width - it looks like I should multiple pixels to 0.234, but maybe there is another, more right way?

**Comment by Pankraty** [comment]
Like many other tasks, this one turns out to be more complicated than it looked initially.

First, we add `ColumnWidthOffset` to the specified column width on saving which may produce an unexpected result.

**Example:**
```
using (var ms = new MemoryStream())
{
    using (var wb = new XLWorkbook())
    {
        var ws = wb.Worksheets.Add("Sheet1");
        Console.WriteLine($"{ws.ColumnWidth}, expected {8.43}");
        wb.SaveAs(ms);
    }
    using (var wb = new XLWorkbook(ms))
    {
        var ws = wb.Worksheet("Sheet1");
        ws.ColumnWidth += 1;
        Console.WriteLine($"{ws.ColumnWidth}, expected {9.43}");
        wb.SaveAs(ms);
    }
    using (var wb = new XLWorkbook(ms))
    {
        var ws = wb.Worksheet("Sheet1");
        ws.ColumnWidth -= 1;
        Console.WriteLine($"{ws.ColumnWidth}, expected {8.43}");
        wb.SaveAs(ms);
    }
}
```
The output:
```
8,43, expected 8,43
9,43, expected 9,43
9,140625, expected 8,43
```
This most likely causes #297 too.

Then, this `ColumnOffsetWidth` is defined as a constant value (0.710625) while in fact, it should depend on a worksheet style because originally it is defined in pixels, and the pixel size depends on the font name and size. More information [here](https://msdn.microsoft.com/en-us/library/office/documentformat.openxml.spreadsheet.column.aspx).

What is strange is that Excel behaves differently for numbers of different length. Say, for Calibri, 11pt and text length from 1 char to 11 it chooses column widths 1.29, 2.29, ... 11.29 respectively (11 digits + 2 pixels which give 11.29), for strings 12-15 digits long it add 3 pixels (12.43, 13.43...), for a string containing 26 digits it sets column width 26.86 (therefore adding 6 pixels), and so on. This behavior looks strange and I don't think it should be imitated in ClosedXML. But we should respect a worksheet style when determining the column width offset.

This said, I think this issue must be addressed after #838 is fixed.

**Comment by igitur** [comment]
About #297, I started looking into it and found a few used posts on this blog http://polymathprogrammer.com/2012/11/18/calculate-excel-column-width-pixel-interval/

I think there was another post on the same blog that seemed especially useful. 

But then exams came and I lost focus on that issue. Obviously time to revisit.
